presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunner.java

```diff
@@ -0,0 +1,301 @@
+    }
+
+    @Override
+    public Optional<EventListener> getEventListener()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public TestingAccessControlManager getAccessControl()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public MaterializedResult execute(String sql)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public MaterializedResult execute(Session session, String sql, List<? extends Type> resultTypes)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<QualifiedObjectName> listTables(Session session, String catalog, String schema)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean tableExists(Session session, String table)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void installPlugin(Plugin plugin)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void createCatalog(String catalogName, String connectorName, Map<String, String> properties)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void loadFunctionNamespaceManager(String functionNamespaceManagerName, String catalogName, Map<String, String> properties)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Lock getExclusiveLock()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getNodeCount()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Session getDefaultSession()
+    {
+        return testSessionBuilder()
+                .setCatalog(catalog)
+                .setSchema(schema)
+                .build();
+    }
+
+    @Override
+    public MaterializedResult execute(Session session, String sql)
+    {
+        String[] command = {
+                "/opt/presto-cli",
+                "--server",
+                "presto-coordinator:" + coordinatorPort,
+                "--catalog",
+                catalog,
+                "--schema",
+                schema,
+                "--execute",
+                sql
+        };
+
+        Container.ExecResult execResult;
+        try {
+            execResult = coordinator.execInContainer(command);
+
+            if (execResult.getExitCode() != 0) {
+                String errorDetails = String.format("Stdout: %s%nStderr: %s", execResult.getStdout(), execResult.getStderr());
+                fail("Presto CLI exited with error code: " + execResult.getExitCode() + "\n" + errorDetails);
+            }
+
+            return ContainerQueryRunnerUtils.toMaterializedResult(execResult.getStdout());
+        }
+        catch (IOException | InterruptedException e) {
+            fail("Execute failed with error message: " + e.getMessage());
```

Comment:
Rethrow the `IOException` as `UncheckedIOException`.  For `InterruptedException`, just rethrow as `RuntimeException`.